midealocal/devices/ac/__init__.py

```diff
@@ -334,7 +336,7 @@ def set_attribute(self, attr: str, value: Any) -> None:
                     )
                     setattr(message, str(self._fresh_air_version), fresh_air)
             elif attr in self._attributes.keys():
-                message = self.make_message_uniq_set()
+                message = self.make_message_set()

```

Comment:
Is this safe?

Reply:
Yes, following the code `make_message_uniq_set()` it just calls 2 different methods based on a flag.
One of the 2 doesn't have the needed fields here, so...

Reply:
But the following code with needed fields is in `if` block...

Reply:
You need a `MessageGeneralSet` object or it won't have the needed fields.
And the original method returns `MessageSubProtocolSet | MessageGeneralSet` based on the if.

So is safe to use always the method that returns  `MessageGeneralSet`

Reply:
If not enter the following `if`,  `MessageSubProtocolSet` seems also ok?

Reply:
Nope.
Check the following fields:

![image](https://github.com/rokam/midea-local/assets/57354320/b81255d5-813e-496b-9339-e9e0893917af)

those are available only for `MessageGeneralSet` ;-)

Reply:
Yes, I mean, if `self._used_subprotocol` is `True` and `attr` not in these attributes,  is this possible?

Reply:
I didn't go too deep into the code, just asked questions about this control flow.
If this situation is impossible, then there is no problem.

Reply:
Ok, I can give approval, but I still think this could be a change

Reply:
https://github.com/rokam/midea-local/pull/33#issuecomment-2254546944